src/core/server/saved_objects/migrations/model/helpers.test.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { addFiltersToQuery } from './helpers';
+
+describe('addFiltersToQuery', () => {
+  it('updates the provided ES query by appending the desired filters to it', () => {
```

Comment:
would be good to add a test where the query already contains a `filter` clause like when `addFiltersToQuery` is called more than once on the same query. 

I suspect the behaviour isn't quite working the way it should when the model calls addFiltersToQuery in `CHECK_UNKNOWN_DOCUMENTS` and then a second time in `CALCULATE_EXCLUDE_FILTERS` because then we would have nested `filter` clauses (maybe it's still a valid ES query?).